vmdb/app/models/ems_refresh/save_inventory.rb

```diff
@@ -17,7 +17,7 @@ def save_vms_inventory(ems, hashes, target = nil)
     log_header = "MIQ(#{self.name}.save_vms_inventory) EMS: [#{ems.name}], id: [#{ems.id}]"
 
     disconnects = if target.kind_of?(ExtManagementSystem) || target.kind_of?(Host)
-      target.vms_and_templates(true).dup
+      target.vms_and_templates(true).to_a.dup
```

Comment:
Will we?  As soon as the association is treated as a list, it should get loaded.  This seemed like a case where it wouldn't get treated like a list until it was too late.